### PR TITLE
Add Tracking to TextView and CocoaTextField

### PR DESCRIPTION
**Pull request by awu-nearside**
- Add tracking property to TextView and CocoaTextField

### CODE REVIEW

**Comment by awu-nearside** [comment]
I'm not too sure how to comment this PR. Should we still include all the points from https://www.notion.so/nearside/Code-Review-Checklist-ad25682f1d5a4888bea40d0dcf26ad99?

File: Sources/Intramodular/Text/TextView.swift [hunk 1]
```diff
@@ -228,6 +229,10 @@ extension _TextView: UIViewRepresentable {
                     if let kerning = configuration.kerning {
                         attributes[.kern] = kerning
                     }
+
+                    if #available(iOS 14.0, *), let tracking = configuration.tracking{
```
Comment (Coledunsby) [inline Sources/Intramodular/Text/TextView.swift:233]:
```suggestion
                    if #available(iOS 14.0, *), let tracking = configuration.tracking {
```

**Review by Coledunsby — APPROVED** [review]
Any idea if tracking overrides kerning in UIKit? I know the SwiftUI docs say
> If you add both the [tracking(_:)](doc://com.apple.documentation/documentation/swiftui/text/tracking(_:)?language=swift) and [kerning(_:)](doc://com.apple.documentation/documentation/swiftui/text/kerning(_:)?language=swift) modifiers to a view, the view applies the tracking and ignores the kerning.

Not a big deal. If that was the case we could have an if/else statement for tracking & kerning to make that more obvious.

**Review by Coledunsby — APPROVED** [review]

**Comment by awu-nearside** [comment]
> Any idea if tracking overrides kerning in UIKit? I know the SwiftUI docs say
> 
> > If you add both the tracking(_:) and kerning(_:) modifiers to a view, the view applies the tracking and ignores the kerning.
> 
> Not a big deal. If that was the case we could have an if/else statement for tracking & kerning to make that more obvious.

I think it does, good point. I'll add in a check that only applies tracking if kerning is not defined.

**Comment by Coledunsby** [comment]
> > Any idea if tracking overrides kerning in UIKit? I know the SwiftUI docs say
> > > If you add both the tracking(_:) and kerning(_:) modifiers to a view, the view applies the tracking and ignores the kerning.
> > 
> > 
> > Not a big deal. If that was the case we could have an if/else statement for tracking & kerning to make that more obvious.
> 
> I think it does, good point. I'll add in a check that only applies tracking if kerning is not defined.

I think we want the other way around – only apply kerning if tracking is not defined... since tracking overrides kerning.

File: Sources/Intramodular/Text/CocoaTextField.swift [hunk 2]
```diff
@@ -228,9 +229,13 @@ fileprivate struct _CocoaTextField<Label: View>: UIViewRepresentable {
...
             uiView.tintColor = context.environment.tintColor?.toUIColor()
             
-            if let kerning = configuration.kerning {
+            if configuration.tracking == nil, let kerning = configuration.kerning {
                 uiView.defaultTextAttributes.updateValue(kerning, forKey: .kern)
             }
+
+            if #available(iOS 14.0, *), let tracking = configuration.tracking {
+                uiView.defaultTextAttributes.updateValue(tracking, forKey: .tracking)
+            }
```
Comment (Coledunsby) [inline Sources/Intramodular/Text/CocoaTextField.swift:238]: we should just make this an if/else statement

File: Sources/Intramodular/Text/TextView.swift [hunk 3]
```diff
@@ -225,9 +226,13 @@ extension _TextView: UIViewRepresentable {
...
                     ]
                     
-                    if let kerning = configuration.kerning {
+                    if configuration.tracking == nil, let kerning = configuration.kerning {
                         attributes[.kern] = kerning
                     }
+
+                    if #available(iOS 14.0, *), let tracking = configuration.tracking {
+                        attributes[.tracking] = tracking
+                    }
```
Comment (Coledunsby) [inline Sources/Intramodular/Text/TextView.swift:235]: same here

**Review by Coledunsby — CHANGES_REQUESTED** [review]

File: Sources/Intramodular/Text/CocoaTextField.swift [hunk 4]
```diff
@@ -228,8 +229,10 @@ fileprivate struct _CocoaTextField<Label: View>: UIViewRepresentable {
             uiView.textContentType = configuration.textContentType
             uiView.tintColor = context.environment.tintColor?.toUIColor()
             
-            if let kerning = configuration.kerning {
+            if configuration.tracking == nil, let kerning = configuration.kerning {
                 uiView.defaultTextAttributes.updateValue(kerning, forKey: .kern)
+            } else if #available(iOS 14.0, *), let tracking = configuration.tracking {
```
Comment (Coledunsby) [inline Sources/Intramodular/Text/CocoaTextField.swift:234]: if we swap these (if tracking ... else if kerning ...) then we don't need the `configuration.tracking == nil` check. Same for the other one below.

Comment (awu-nearside) [reply Sources/Intramodular/Text/CocoaTextField.swift:234]: oops, you're right.

File: Sources/Intramodular/Text/CocoaTextField.swift [hunk 5]
```diff
@@ -229,7 +229,7 @@ fileprivate struct _CocoaTextField<Label: View>: UIViewRepresentable {
             uiView.textContentType = configuration.textContentType
             uiView.tintColor = context.environment.tintColor?.toUIColor()
             
-            if configuration.tracking == nil, let kerning = configuration.kerning {
+            if let kerning = configuration.kerning {
```
Comment (Coledunsby) [inline Sources/Intramodular/Text/CocoaTextField.swift:232]: I believe tracking should have priority so we should swap the if & else cases

**Review by Coledunsby — APPROVED** [review]

**Pull request merged**